apps/web/src/modules/dashboard/DaoProposalCard.tsx

```diff
@@ -85,3 +95,54 @@ export const DaoProposalCard = ({
     </Link>
   )
 }
+
+type NeedsVoteProps = {
+  userAddress?: AddressType
+  chainId: CHAIN_ID
+  proposalId: string
+  proposalState: ProposalState
+}
+const NeedsVote = ({
+  userAddress,
+  chainId,
+  proposalState,
+  proposalId,
+}: NeedsVoteProps) => {
+  const [showTooltip, setShowTooltip] = useState(false)
+  const { data: hasVoted } = useSWR(
+    [proposalId, userAddress, chainId],
+    async () => {
+      if (proposalState !== ProposalState.Active) return null
+      const hasVoted = await hasUserVoted(proposalId, userAddress as AddressType, chainId)
```

Comment:
is there a way we can query this with the aggregate query? seems inefficient to query this for every single proposal